Represent ImageOnlyFailures with a separate icon when looking at results on the databases

robert-jenner opened the pull request:
#### c46ef76f2def121dc6357786c87bd57fbef8dceb
<pre>
Represent ImageOnlyFailures with a separate icon when looking at results on the databases
<a href="https://bugs.webkit.org/show_bug.cgi?id=248911">https://bugs.webkit.org/show_bug.cgi?id=248911</a>
rdar://102920285

Reviewed by NOBODY (OOPS!).

It would be easier if ImageOnlyFailures were marked with a different
icon from text failures. The changes in this commit are to effectivly
mark ImageOnlyFailures with a blue I instead of a red X. And the colour
blue has been specifically chosen as one that is colour blind friendly.

* Tools/Scripts/libraries/resultsdbpy/resultsdbpy/view/static/js/expectations.js:
(Expectations.colorMap):
(Expectations.unexpectedResults):
(Expectations):
* Tools/Scripts/libraries/resultsdbpy/resultsdbpy/view/static/library/css/webkit.css:
(:root):
</pre>

JonWBedard commented on Tools/Scripts/libraries/resultsdbpy/resultsdbpy/view/static/js/expectations.js:
```diff
@@ -62,7 +63,7 @@ class Expectations
                 r.splice(i, 1);
             if (expectation === 'FAIL')
-                ['TEXT', 'AUDIO', 'IMAGE'].forEach(expectation => {
+                ['TEXT', 'AUDIO'].forEach(expectation => {
```
I don't think we want to take `IMAGE` out here. What this bit of code does is let a test marked as expected to `FAIL` show a green check if that test actually has an `IMAGE` failure, which I think should still be true after your change, right?

robert-jenner replied:
I think so. Let me add it back.

JonWBedard commented on Tools/Scripts/libraries/resultsdbpy/resultsdbpy/view/static/js/expectations.js:
```diff
@@ -88,10 +89,11 @@ Expectations.stateToIdMap = {
     PASS: 0x40,
 };
-Expectations.failureTypes = ['warning', 'failed', 'timedout', 'crashed'];
+Expectations.failureTypes = ['warning', 'failed', 'image', 'timedout', 'crashed'];
```
I think we need to update the legend in `Tools/Scripts/libraries/resultsdbpy/resultsdbpy/view/static/js/timeline.js` too

robert-jenner replied:
Got it. I will work on that. Re-submit and re-stage.

robert-jenner committed:
Represent ImageOnlyFailures with a separate icon when looking at results on the databases
https://bugs.webkit.org/show_bug.cgi?id=248911
rdar://102920285

Reviewed by NOBODY (OOPS!).

It would be easier if ImageOnlyFailures were marked with a different
icon from text failures. The changes in this commit are to effectivly
mark ImageOnlyFailures with a blue I instead of a red X. And the colour
blue has been specifically chosen as one that is colour blind friendly.

* Tools/Scripts/libraries/resultsdbpy/resultsdbpy/view/static/js/expectations.js:
(Expectations.colorMap):
(Expectations.unexpectedResults):
(Expectations):
* Tools/Scripts/libraries/resultsdbpy/resultsdbpy/view/static/library/css/webkit.css:
(:root):
